From: Khem Raj <raj.khem@gmail.com>
Subject: [PATCH 03/12] ltmain.in: Add missing sysroot to library path

When using a sysroot we should append it to libdir, which is helpful in
cross builds as the system is staged in the sysroot. For normal builds,
i.e. when lt_sysroot is not set, it will still behave the same and add
-L/usr/lib to the relink command.

Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

Submitted [https://lists.gnu.org/archive/html/libtool-patches/2021-10/msg00017.html]
Upstream-Status: Backport [https://git.savannah.gnu.org/cgit/libtool.git/commit/?id=3221f9f0fb98d5740ab5d0e8db6a731302520644]

Index: libtool-2.5.0/build-aux/ltmain.in
===================================================================
--- libtool-2.5.0.orig/build-aux/ltmain.in
+++ libtool-2.5.0/build-aux/ltmain.in
@@ -6514,7 +6514,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      add_dir="-L$lt_sysroot$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
